internal/record/recording_https_proxy.go

```diff
@@ -178,17 +161,40 @@ func (r *RecordingHTTPSProxy) proxyRequest(w http.ResponseWriter, req *http.Requ
 	return resp, respBodyBytes, nil
 }
 
-func (r *RecordingHTTPSProxy) recordResponse(resp *http.Response, fileName string, body []byte) error {
+func (r *RecordingHTTPSProxy) recordResponse(recReq *store.RecordedRequest, resp *http.Response, fileName string, body []byte) error {
 	recordedResponse, err := store.NewRecordedResponse(resp, body)
 	if err != nil {
 		return err
 	}
+	recordPath := filepath.Join(r.recordingDir, fileName+".http.log")
+	shaSum := recReq.ComputeSum()
 
-	recordedResponse.Body = r.redactor.Bytes(recordedResponse.Body)
+	// Open the file in append mode, create it if it doesn't exist.
+	file, err := os.OpenFile(recordPath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	fmt.Printf("Writing request to: %s\n", recordPath)
+	serializedReq := recReq.Serialize()
+	_, err = file.WriteString(fmt.Sprintf("%s.req %d\n", shaSum, len(serializedReq)))
+	if err != nil {
+		return err
+	}
+	_, err = file.WriteString(serializedReq)
+	if err != nil {
+		return err
+	}
 
-	recordPath := filepath.Join(r.recordingDir, fileName+".resp")
 	fmt.Printf("Writing response to: %s\n", recordPath)
-	err = os.WriteFile(recordPath, []byte(recordedResponse.Serialize()), 0644)
+    recordedResponse.Body = r.redactor.Bytes(recordedResponse.Body)
+	serializedResp := recordedResponse.Serialize()
+	_, err = file.WriteString(fmt.Sprintf("\n%s.resp %d\n", shaSum, len(serializedResp)))
```

Comment:
Why do we need to write `<shaSum>.resp` here?

Reply:
I use `<shaSum>.resp` as delimiter to parse the replay response from the replay file

see an example replay log https://gist.github.com/hkt74/1ffa7b17bcf45b33d8b1a9d806a69784